### PR TITLE
Frontend: Modal component

### PR DESCRIPTION
**Pull request by kobim**
A connected Modal component which will be used for an "Add a Restaurant" screen.

### CODE REVIEW

File: client/src/components/Modal/redux/selectors.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { createSelector } from 'reselect';
+import { get } from 'lodash';
+
+const getModals = state => state.modals;
+
+export const getModalName = (_, props) => props.name;
+
+export const getModalState = createSelector(
+  [getModals, getModalName],
+  (modals, name) => get(modals, name, false),
```
Comment (eladitzhakian) [inline client/src/components/Modal/redux/selectors.js]: I wouldn't add `lodash` as a dependency just for this simple use case.
Alternatives:
```
modals[name] || false
// or
!!modals[name]
```

File: client/src/scss/Modal.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
+@import 'vars';
+
+.modal {
+  position: fixed;
+  display: none;
+  z-index: 999;
```
Comment (eladitzhakian) [inline client/src/scss/Modal.scss]: This works but I suggest not to make a habit of it 💃

File: client/src/components/Header/index.test.jsx [hunk 3]
```diff
@@ -2,26 +2,44 @@ import React from 'react';
...
-    const wrapper = shallow(<Header />);
+    const wrapper = shallow(<Header {...props} />);
 
     expect(wrapper.find('h1.title').text()).toBe('WeEat');
     expect(wrapper.find('h2.title').text()).toContain('12:00');
   });
 
-  it('has a search bar', () => {
-    const wrapper = shallow(<Header />);
+  it('has a search bar and an add button', () => {
```
Comment (NoyaArie) [inline client/src/components/Header/index.test.jsx]: `and an add button`?

File: client/src/components/Modal/index.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { setModalState } from './redux';
+import { getModalState } from './redux/selectors';
+
+import '../../scss/Modal.scss';
+
+export const Modal = ({ name, open, setModal, children }) => (
+  <div className={`modal ${ open ? 'modal-open' : '' }`}>
```
Comment (eladitzhakian) [inline client/src/components/Modal/index.jsx]: For these cases I prefer using the [classnames](https://www.npmjs.com/package/classnames) package;
```
import cn from 'classnames';

<div className={cn('modal', { 'modal-open': open })} />
```

File: client/src/components/Modal/index.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { setModalState } from './redux';
+import { getModalState } from './redux/selectors';
+
+import '../../scss/Modal.scss';
```
Comment (yanivmichaelis) [inline client/src/components/Modal/index.jsx]: this is a bit different between teams and projects - but keeping the style next to the component, in the same directory is quite convenient

File: client/src/components/Modal/redux/selectors.js (hunk 1 above)
Comment (kobim) [reply client/src/components/Modal/redux/selectors.js]: _lodash_ is coming heavily in next PRs :)

File: client/src/scss/Modal.scss (hunk 2 above)
Comment (kobim) [reply client/src/scss/Modal.scss]: 😇

File: client/src/components/Modal/index.jsx (hunk 4 above)
Comment (yanivmichaelis) [inline client/src/components/Modal/index.jsx]: we use this lib to help out with classnames: https://github.com/JedWatson/classnames

File: client/src/components/Modal/index.jsx (hunk 5 above)
Comment (kobim) [reply client/src/components/Modal/index.jsx]: 👍

File: client/src/scss/Modal.scss (hunk 2 above)
Comment (yanivmichaelis) [inline client/src/scss/Modal.scss]: 🤢

**Pull request merged**